### DISCUSSION TITLE
Is there any way to make prisma do only one query when querying from two tables?

### DISCUSSION THREAD

**Discussion by mishase** [prisma/prisma#11738] [Q&A] [answered]
## First attempt
```ts
await prisma.session.findUnique({
	select: {
		user: { select: { firstName: true } },
	},
	where: {
		token,
	},
});
```
`SELECT "public"."Session"."token", "public"."Session"."userId" FROM "public"."Session" WHERE "public"."Session"."token" = $1 LIMIT $2 OFFSET $3`
`SELECT "public"."User"."id", "public"."User"."firstName" FROM "public"."User" WHERE "public"."User"."id" IN ($1) OFFSET $2`
## Second attempt
```ts
await prisma.session.findUnique({
	include: {
		user: { select: { firstName: true } },
	},
	where: {
		token,
	},
});
```
`SELECT "public"."Session"."token", "public"."Session"."userId", "public"."Session"."createdAt" FROM "public"."Session" WHERE "public"."Session"."token" = $1 LIMIT $2 OFFSET $3`
`SELECT "public"."User"."id", "public"."User"."firstName" FROM "public"."User" WHERE "public"."User"."id" IN ($1) OFFSET $2`
## Can be replaced by one query
`SELECT "User"."firstName" FROM "User" LEFT JOIN "Session" ON "User".id = "Session"."userId" WHERE "Session".token = ${token}`

**Comment by pantharshit00** [comment, marked as answer]
This is currently not possible since prisma doesn't use joins under the hood. There is an open issue about this here: https://github.com/prisma/prisma/issues/5184

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
